src/rocksdb.cc

```diff
@@ -1441,3 +1442,79 @@ rocksdb_compact_range(rocksdb_t *db, rocksdb_compact_range_t *req, rocksdb_colum
 
   return uv_queue_work(req->req.db->loop, &req->req.worker, rocksdb__on_compact_range, rocksdb__on_after_compact_range);
 }
+
+namespace {
+
+static void
+rocksdb__on_after_approximate_size(uv_work_t *handle, int status) {
+  auto req = reinterpret_cast<rocksdb_approximate_size_t *>(handle->data);
+
+  rocksdb__remove_req(req);
+
+  auto error = req->error;
+
+  req->cb(req, status);
+
+  if (error) free(error);
+}
+
+static void
+rocksdb__on_approximate_size(uv_work_t *handle) {
+  auto req = reinterpret_cast<rocksdb_approximate_size_t *>(handle->data);
+
+  auto db = reinterpret_cast<DB *>(req->req.db->handle);
+
+  auto column_family = reinterpret_cast<ColumnFamilyHandle *>(req->column_family);
+
+  auto start = rocksdb__slice_cast(req->start);
+  auto end = rocksdb__slice_cast(req->end);
+
+  SizeApproximationOptions options;
+
+  options.include_memtables = true;
+  options.files_size_error_margin = 0.1;
+
+  constexpr int SIZE = 1;
+
+  std::array<Range, SIZE> ranges;
+  std::array<uint64_t, SIZE> result;
```

Comment:
This should use plain pointers as there's only a single range and a single result, no need for `std::array`.